### PR TITLE
[inetstack] Allow `TcpSegment`'s `data` Field to be Optional

### PR DESCRIPTION
**Pull request by BrianZill**
This PR changes the `TcpSegment` struct so that the `data` field is now `Option<Buffer>` instead of `Buffer`.  This struct is used in issuing transmission requests to the runtime.  Before this change, sending a header-only TCP segment required one to create an empty `Buffer` just to be able to have something to fill this field.  When `Buffer` was a pure stack object, this was still somewhat acceptable, now that we're planning on having all `Buffer`s live in the heap, this change will save us from needless allocations.

This PR fixes Issue #202.

### CODE REVIEW

File: src/protocols/tcp/established/ctrlblk.rs [hunk 1]
```diff
@@ -809,14 +809,19 @@ impl ControlBlock {
...
     ///
-    pub fn emit(&self, header: TcpHeader, data: Buffer, remote_link_addr: MacAddress) {
-        debug!("Sending {} bytes + {:?}", data.len(), header);
+    pub fn emit(&self, header: TcpHeader, body: Option<Buffer>, remote_link_addr: MacAddress) {
+        // ToDo: Would prefer to only perform this if check in debug builds.
+        if body.is_some() {
+            debug!("Sending {} bytes + {:?}", body.as_ref().unwrap().len(), header);
+        } else {
+            debug!("Sending 0 bytes + {:?}", header);
+        }
```
Comment (ppenna) [inline src/protocols/tcp/established/ctrlblk.rs:824]: Me too :)

Why don't you use [`debug_assertions`](https://stackoverflow.com/questions/39204908/how-to-check-release-debug-builds-using-cfg-in-rust) ?

**Review by ppenna — COMMENTED** [review]
Looks good. I just add a suggestion on how to get that condition to be evaluated only on debug builds.

File: src/protocols/tcp/established/ctrlblk.rs (hunk 1 above)
Comment (BrianZill) [reply src/protocols/tcp/established/ctrlblk.rs:824]: Interesting, thanks!  I figured there had to be some way to do this in Rust.

Comment (BrianZill) [reply src/protocols/tcp/established/ctrlblk.rs]: Fixed.

**Review by ppenna — APPROVED** [review]

**Pull request merged**